proving.go

```diff
@@ -2,6 +2,7 @@ package merkle
 
 import (
 	"errors"
+	"io"
```

Comment:
adding comment here for lack of better place - add sanity verification that the cache structure make sense

Reply:
done https://github.com/spacemeshos/merkle-tree/pull/5/commits/aeeb27019da40a73462e70e5dd16e0511b52ccd2